Tiltfile

```diff
@@ -0,0 +1,201 @@
+default_registry(settings.get("default_registry"))
+local_goos = settings.get("local_goos", )
+
+tilt_helper_dockerfile_header = """
+# Tilt image
+FROM golang:1.15.3 as tilt-helper
+# Support live reloading with Tilt
+RUN wget --output-document /restart.sh --quiet https://raw.githubusercontent.com/windmilleng/rerun-process-wrapper/master/restart.sh  && \
+    wget --output-document /start.sh --quiet https://raw.githubusercontent.com/windmilleng/rerun-process-wrapper/master/start.sh && \
+    chmod +x /start.sh && chmod +x /restart.sh
+"""
+
+additional_docker_helper_commands = """
+RUN wget -qO- https://dl.k8s.io/v1.19.2/kubernetes-client-linux-amd64.tar.gz | tar xvz
+RUN wget -qO- https://get.docker.com | sh
+"""
+
+additional_docker_build_commands = """
+COPY --from=tilt-helper /usr/bin/docker /usr/bin/docker
+COPY --from=tilt-helper /go/kubernetes/client/bin/kubectl /usr/bin/kubectl
+"""
+
+##############################
+# Setup Velero
+##############################
+
+# Set up a local_resource build of the Velero binary. The binary is written to _tiltbuild/velero.
+local_resource(
+    "velero_server_binary",
+    cmd = 'cd ' + '.' + ';mkdir -p _tiltbuild;PKG=. BIN=velero GOOS=linux GOARCH=amd64 VERSION=main GIT_TREE_STATE=dirty OUTPUT_DIR=_tiltbuild ./hack/build.sh',
+    deps = ["cmd", "internal", "pkg"],
+    ignore = ["pkg/cmd"],
+)
+
+local_resource(
+    "velero_local_binary",
+    cmd = 'cd ' + '.' + ';mkdir -p _tiltbuild/local;PKG=. BIN=velero GOOS=' + local_goos + ' GOARCH=amd64 VERSION=main GIT_TREE_STATE=dirty OUTPUT_DIR=_tiltbuild/local ./hack/build.sh',
+    deps = ["internal", "pkg/cmd"],
+)
+
+tilt_dockerfile_header = """
+FROM gcr.io/distroless/base:debug as tilt
+WORKDIR /
+COPY --from=tilt-helper /start.sh .
+COPY --from=tilt-helper /restart.sh .
+COPY velero .
+"""
+
+dockerfile_contents = "\n".join([
+    tilt_helper_dockerfile_header,
+    additional_docker_helper_commands,
+    tilt_dockerfile_header,
+    additional_docker_build_commands,
+])
+
+ # Set up an image build for Velero. The live update configuration syncs the output from the local_resource
+ # build into the container.
+docker_build(
+    ref = "velero/velero",
+    context = "_tiltbuild/",
+    dockerfile_contents = dockerfile_contents,
+    target = "tilt",
+    entrypoint=["sh", "/start.sh", "/velero"],
+    live_update=[
+        sync("./_tiltbuild/velero", "/velero"),
+        run("sh /restart.sh"),
+    ])
+
+##############################
+# Setup plugins
+##############################
+
+providers = {}
+
+def load_provider_tiltfiles():
+    provider_repos = settings.get("provider_repos", [])
+
+    for repo in provider_repos:
+        file = repo + "/tilt-provider.json"
+        provider_details = read_json(file, default = {})
+        if type(provider_details) != type([]):
+            provider_details = [provider_details]
+        for item in provider_details:
+            provider_name = item["name"]
```

Comment:
When the `tilt-provider.json` file is missing, Tilt will report an error like this:

```
Beginning Tiltfile execution
Falling back to `kubectl kustomize` since `kustomize` was not found in PATH
Successfully loaded Tiltfile (172.487237ms)
Traceback (most recent call last):
  /Users/nrb/go/src/github.com/vmware-tanzu/velero/Tiltfile:199:24: in <toplevel>
  /Users/nrb/go/src/github.com/vmware-tanzu/velero/Tiltfile:126:33: in load_provider_tiltfiles
Error: key "name" not in dict
```

I'd recommend adding the following after L21:

```python
        if not os.path.exists(file):
            print("No provider.json found for %s", repo)
            continue
```

That will then lead to this error:

```
Beginning Tiltfile execution
Falling back to `kubectl kustomize` since `kustomize` was not found in PATH
No provider.json found for %s ../velero-plugin-for-gcp
Successfully loaded Tiltfile (74.249484ms)
Traceback (most recent call last):
  /Users/nrb/go/src/github.com/vmware-tanzu/velero/Tiltfile:204:17: in <toplevel>
  /Users/nrb/go/src/github.com/vmware-tanzu/velero/Tiltfile:143:24: in enable_providers
  /Users/nrb/go/src/github.com/vmware-tanzu/velero/Tiltfile:151:20: in enable_provider
Error: NoneType has no .get field or method
```

I have a fix for it, but I'll add it inline where it goes.